Froggie/Source/Froggie.Data/Groups/GetUsersGroupsQuery.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Froggie.Domain.Groups;
+using Froggie.Domain.Users;
+using LittleByte.Common.AspNet.AutoMapper;
+
+namespace Froggie.Data.Groups;
+
+internal sealed class GetUsersGroupsQuery : IGetUsersGroupsQuery
+{
+    private readonly FroggieDb froggieDb;
+    private readonly IMapper mapper;
+
+    public GetUsersGroupsQuery(FroggieDb froggieDb, IMapper mapper)
+    {
+        this.froggieDb = froggieDb;
+        this.mapper = mapper;
+    }
+
+    public async ValueTask<IReadOnlyCollection<Group>> QueryAsync(Id<User> userId)
+    {
+        var daos = await froggieDb
+            .UserGroupMaps.AsNoTracking()
```

Comment:
The persistence framework we are using is called Entity Framework Core. By default (ie: without `AsNoTracking`) it will keep a reference to classes it loads from a database so it can watch for changes to that object. `AsNoTracking`  tells EFC to not worry about watching the object. Generally this is done for performance reasons when you are loading an object just to read it (such as to send it to the client asking for it) since there will be no changes made.

However there is another reason to use `AsNoTracking`. Since we are using a 3 layer data moves between them (persistence <-> domain <-> dto). When we move data between layers we create a new object. Example:

1. Edit task request comes in
2. Task is loaded from database as DAO (this is what EFC holds a reference to)
3. We convert it to a domain model right away to ensure changes are always done in the domain layer under our business rules
4. After the changes are successfully applied we convert the domain model into a new DAO (EFC has no knowledge of this object)
5. When we tell EFC to update the task with this new DAO it throws an error saying that an object with that ID already exists